src/openrct2/rct2/S6Exporter.cpp

```diff
@@ -152,6 +152,7 @@ void S6Exporter::Save(IStream * stream, bool isScenario)
 
 void S6Exporter::Export()
 {
+    openrct2_assert(!(check_for_spatial_index_cycles(false) || check_for_sprite_list_cycles(false)), "A sprite cycle exists.");
```

Comment:
Could it say where the cycle exists? E.g. `A sprite cycle exists in the spatial index`